lib/citeproc/ruby/renderer/date.rb

```diff
@@ -30,56 +28,91 @@ def render_date(item, node)
           parts, delimiter = node.parts, node.delimiter
         end
 
-        parts.map { |part|
-          render date, part
+        if date.range?
+          render_date_range date, node, parts, delimiter
+        else
+          parts.map { |part|
+            render date, part
+          }.reject(&:empty?).join(delimiter)
+        end
+      end
+
+      def render_date_range(date, node, parts, delimiter)
+        delimit_range_at = 'day' # TODO date.delimit_range_at
```

Comment:
This should be implemented in `CiteProc::Date` to return the he largest date part (“year”, “month”, or “day”) that differs between the two dates.